File: sdk/generator/python/template/polar/version/webhooks.py
from __future__ import annotations

import dataclasses
import typing

{% if enum_imports %}
from polar.{{ version }}.literals import (
{% for enum_name in enum_imports %}
    {{ enum_name }},
{% endfor %}
)
{% endif %}
{% if imports %}
from polar.{{ version }}.outputs import (
{% for name in imports %}
    {{ name }},
{% endfor %}
)
{% endif %}

{% for model in api.webhooks %}
@dataclasses.dataclass(kw_only=True, slots=True)
class {{ model.name }}:
{% if model.description %}
    """{{ model.description }}"""
{% endif %}
{% for field in model.fields %}
    {% if field.default is not none %}
    {{ field.name }}: {{ field.type | type_annotation }} = {{ field.default | format_default_dataclass }}
    {% elif not field.required %}
    {{ field.name }}: {{ field.type | wrap_nullable | type_annotation }} = None
    {% else %}
    {{ field.name }}: {{ field.type | type_annotation }}
    {% endif %}
{% if field.description %}
    """{{ field.description }}"""
{% endif %}

{% else %}
    ...
{% endfor %}
{% endfor %}

{% if api.webhooks %}
WebhookPayload: typing.TypeAlias = (
{% for model in api.webhooks %}
    {{ model.name }}{% if not loop.last %} |{% endif %}
{% endfor %}
)
{% else %}
WebhookPayload: typing.TypeAlias = typing.Never
{% endif %}

File: sdk/generator/typescript/template/src/version/webhooks.ts
{% if imports %}import type {
{% for name in imports %}  {{ name }},
{% endfor %}} from "./models";

{% endif %}{% for model in api.webhooks %}/**
{{ (model.description or model.name) | format_description }}
 */
export interface {{ model.name }}{% if model.fields %} {
  {% for field in model.fields %}  /**
{{ (field.description or field.name) | format_description | indent(2, true) }}
   */
  {{ field.name }}{% if not field.required %}?{% endif %}: {{ field.type | ts_type }};
  {% endfor %}}{% else %} extends Record<string, never> {}{% endif %}

{% endfor %}export type WebhookPayload = {% if api.webhooks %}{% for model in api.webhooks %}{{ model.name }}{% if not loop.last %} | {% endif %}{% endfor %}{% else %}never{% endif %};
